pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -88,13 +88,22 @@ func Add(mgr manager.Manager, context *controllerconfig.Context) error {
 	}
 
 	pdbPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			// do not reconcile when PDB is created
+			return false
+		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			pdb, ok := e.ObjectNew.DeepCopyObject().(*policyv1beta1.PodDisruptionBudget)
 			if !ok {
 				return false
 			}
-			// only reconcile if allowed disruptions is 0 in the main PDB
-			return pdb.Name == osdPDBAppName && pdb.Status.DisruptionsAllowed == 0
+			// only reconcile if allowed disruptions for the main PDB is 0 in the operator namespace
+			return e.ObjectNew.GetNamespace() == context.OperatorNamespace &&
```

Comment:
What if we just don't check for the namespace? Then we can still check all namespaces in the scenario where the cluster is in a different namespace.

Reply:
I believe that if we don't, we would reconcile on any PDB updates, right @sp98, checking the ns addresses that issue IIUC.

Reply:
Yes, that's right. 

But one more thing is that,  previously, `CreateFunc` and `DeleteFunc` were not no-op.  Now I've added them as no no-op. So only `pdb.Name == osdPDBAppName && pdb.Status.DisruptionsAllowed == 0` should be enough to ensure that we are not reconciling for PDB updates on other ns (unless some other ns has a pdb named `osdPDBAppName` which is not likely). 

 I feel we can still get rid of `e.ObjectNew.GetNamespace() == context.OperatorNamespace` check.

Reply:
Agreed, seems like the pdb name is a sufficient filter so we can still check all namespaces

Reply:
removed namespace check.